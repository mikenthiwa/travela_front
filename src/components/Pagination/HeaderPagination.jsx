import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Utils from '../../helper/Utils';
import './_headerPagination.scss';
import SelectDropDown from '../SelectDropDown/SelectDropDown';

const numberLimits = [
  {
    name: '10',
    value: 10,
  },
  {
    name: '20',
    value: 20,
  },
  {
    name: '30',
    value: 30,
  },
];
class HeaderPagination extends PureComponent {
  render() {
    const { getRequestsWithLimit, url } = this.props;
    const limit = Utils.getCurrentLimit(url);
    return (
      <div className="cell-items-per-page">
        <div className="cell-items-per-page-text">
          <span className="items-per-page-text">
            Items per page
          </span>
        </div>
        <div className="items-per-page-paginator">
          <SelectDropDown
            onClickItem={getRequestsWithLimit}
            dropDownItems={numberLimits}
            defaultSelected={limit}
          />
        </div>
      </div>
    );
  }
}

HeaderPagination.propTypes = {
  getRequestsWithLimit: PropTypes.func,
  url: PropTypes.string,
};
HeaderPagination.defaultProps = {
  getRequestsWithLimit: null,
  url: ''
};

export default HeaderPagination;
