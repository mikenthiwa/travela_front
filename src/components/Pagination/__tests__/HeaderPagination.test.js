import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import HeaderPagination from '../HeaderPagination';

const props = {
  getRequestsWithLimit: sinon.spy(() => Promise.resolve())
};

describe('<HeaderPagination />', () => {
  it('should render successfully', () => {
    const wrapper = shallow(<HeaderPagination {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the `getRequestsWithLimit` method when a limit is selected', () => {
    const wrapper = mount(<HeaderPagination {...props} />);
    const { getRequestsWithLimit } = props;
    const limit = wrapper.find('.dropdown__list__item').first();
    limit.simulate('click');
    expect(getRequestsWithLimit.called).toEqual(true);
    expect(getRequestsWithLimit.calledWith(10)).toEqual(true);
  });
});
