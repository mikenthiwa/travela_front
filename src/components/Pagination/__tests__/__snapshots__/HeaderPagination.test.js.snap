// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderPagination /> should render successfully 1`] = `
<div
  className="cell-items-per-page"
>
  <div
    className="cell-items-per-page-text"
  >
    <span
      className="items-per-page-text"
    >
      Items per page
    </span>
  </div>
  <div
    className="items-per-page-paginator"
  >
    <SelectDropDown
      defaultSelected=""
      dropDownClass=""
      dropDownIcon="drop-down-icon.svg"
      dropDownItems={
        Array [
          Object {
            "name": "10",
            "value": 10,
          },
          Object {
            "name": "20",
            "value": 20,
          },
          Object {
            "name": "30",
            "value": 30,
          },
        ]
      }
      onClickItem={[Function]}
      placeHolder=""
    />
  </div>
</div>
`;
